docs/docker-example/Dockerfile

```diff
@@ -1,19 +1,51 @@
-FROM onnxmlirczar/onnx-mlir-build:amd64
+FROM onnxmlirczar/onnx-mlir-dev
 
-WORKDIR /build
-ENV HOME=/build
-ENV PYENV_ROOT=$HOME/.pyenv
-ENV PATH=$PYENV_ROOT/shims:$PYENV_ROOT/bin:$PATH
-RUN pyenv global 3.7.0
-RUN pyenv rehash
+WORKDIR /workdir
+ENV HOME=/workdir
 
-ENV PATH=$PATH:/build/bin
+# install needed packages
+ENV PATH=$PATH:/workdir/bin
 RUN apt-get update
 RUN apt-get install -y python-numpy
 RUN apt-get install -y python3-pip
+RUN python -m pip install --upgrade pip
 RUN apt-get install -y gdb
 RUN apt-get install -y lldb
 RUN apt-get install -y emacs
-WORKDIR /build/.vscode
-ADD .vscode /build/.vscode
-WORKDIR /build
+RUN apt-get install -y libeigen3-dev
+RUN apt-get install -y clang-format
+# install clang-12
+RUN apt-get install -y lsb-release wget software-properties-common
+RUN bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)"
+RUN python -m pip install wheel
+RUN python -m pip install numpy
+RUN python -m pip install torch==1.6.0+cpu torchvision==0.7.0+cpu -f https://download.pytorch.org/whl/torch_stable.html
+RUN git clone https://github.com/onnx/tutorials.git
+
+
+# set env variables for make - these should be set in any shell that is started
+ENV LLVM_PROJ_SRC=/workdir/llvm-project
+ENV LLVM_PROJ_WORKDIR=/workdir/llvm-project/build
+ENV LLVM_PROJ_BUILD=/workdir/llvm-project/build
+ENV NPROC=4
+
+WORKDIR /workdir/onnx-mlir/build
+
+# copy in vscode configuration info
+WORKDIR /workdir/.vscode
+ADD .vscode /workdir/.vscode
+
+# copy workspace folder (typically contains local/personal testcases etc)
+WORKDIR /workdir/workspace
+ADD workspace /workdir/workspace
+
+# do stuff to fix git
+WORKDIR /workdir/onnx-mlir
+# reattach HEAD
+RUN git checkout master
+# make git see branches other than master
+RUN git config remote.origin.fetch "+refs/heads/*:refs/remotes/origin/*"
+
+WORKDIR /workdir
+ENV PATH=$PATH:/workdir/onnx-mlir/build/Debug/bin/:/workdir/onnx-mlir/build/Debug/lib:/workdir/llvm-project/build/Debug/bin
+
```

Comment:
Tried the script and I got this. Not sure if this is a transient error or not.
```
(onnx-mlir) alexe@pancetta Git % docker build --tag amd64dev-from-git-2021-05-27 .
[+] Building 1.5s (26/30)
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                                                                                                                0.0s
 => => transferring dockerfile: 37B                                                                                                                                                                                                                                                                                                                 0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                                                                                                                   0.0s
 => => transferring context: 2B                                                                                                                                                                                                                                                                                                                     0.0s
 => [internal] load metadata for docker.io/onnxmlirczar/onnx-mlir-dev:latest                                                                                                                                                                                                                                                                        0.2s
 => CANCELED [ 1/26] FROM docker.io/onnxmlirczar/onnx-mlir-dev@sha256:12555c7f09ee203bf32f8457d2672395fa491a03b4b06a500d525052bf155dc1                                                                                                                                                                                                              1.1s
 => => resolve docker.io/onnxmlirczar/onnx-mlir-dev@sha256:12555c7f09ee203bf32f8457d2672395fa491a03b4b06a500d525052bf155dc1                                                                                                                                                                                                                         0.0s
 => => sha256:5001243a28a1a1dc48efc34a8b869f2b56d0cf5343cb964b64377ae5ab0d3179 9.44MB / 4.49GB                                                                                                                                                                                                                                                      1.1s
 => => sha256:12555c7f09ee203bf32f8457d2672395fa491a03b4b06a500d525052bf155dc1 780B / 780B                                                                                                                                                                                                                                                          0.0s
 => => sha256:4f501d71c82bba387b13973c586f870839c7b02bd5ec7bbe0a5d1b8c21b699ce 738B / 738B                                                                                                                                                                                                                                                          0.0s
 => => sha256:8824a461cec06de796e83c608332f0fd5234995182fdcfbeb16f336486750d2c 3.60kB / 3.60kB                                                                                                                                                                                                                                                      0.0s
 => [internal] load build context                                                                                                                                                                                                                                                                                                                   1.1s
 => => transferring context: 2.48MB                                                                                                                                                                                                                                                                                                                 1.1s
 => CACHED [ 2/26] WORKDIR /workdir                                                                                                                                                                                                                                                                                                                 0.0s
 => CACHED [ 3/26] RUN apt-get update                                                                                                                                                                                                                                                                                                               0.0s
 => CACHED [ 4/26] RUN apt-get install -y python-numpy                                                                                                                                                                                                                                                                                              0.0s
 => CACHED [ 5/26] RUN apt-get install -y python3-pip                                                                                                                                                                                                                                                                                               0.0s
 => CACHED [ 6/26] RUN python -m pip install --upgrade pip                                                                                                                                                                                                                                                                                          0.0s
 => CACHED [ 7/26] RUN apt-get install -y gdb                                                                                                                                                                                                                                                                                                       0.0s
 => CACHED [ 8/26] RUN apt-get install -y lldb                                                                                                                                                                                                                                                                                                      0.0s
 => CACHED [ 9/26] RUN apt-get install -y emacs                                                                                                                                                                                                                                                                                                     0.0s
 => CACHED [10/26] RUN apt-get install -y libeigen3-dev                                                                                                                                                                                                                                                                                             0.0s
 => CACHED [11/26] RUN apt-get install -y clang-format                                                                                                                                                                                                                                                                                              0.0s
 => CACHED [12/26] RUN apt-get install -y lsb-release wget software-properties-common                                                                                                                                                                                                                                                               0.0s
 => CACHED [13/26] RUN bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)"                                                                                                                                                                                                                                                                          0.0s
 => CACHED [14/26] RUN python -m pip install wheel                                                                                                                                                                                                                                                                                                  0.0s
 => CACHED [15/26] RUN python -m pip install numpy                                                                                                                                                                                                                                                                                                  0.0s
 => CACHED [16/26] RUN python -m pip install torch==1.6.0+cpu torchvision==0.7.0+cpu -f https://download.pytorch.org/whl/torch_stable.html                                                                                                                                                                                                          0.0s
 => CACHED [17/26] RUN git clone https://github.com/onnx/tutorials.git                                                                                                                                                                                                                                                                              0.0s
 => CACHED [18/26] WORKDIR /workdir/onnx-mlir/build                                                                                                                                                                                                                                                                                                 0.0s
 => CACHED [19/26] WORKDIR /workdir/.vscode                                                                                                                                                                                                                                                                                                         0.0s
 => CACHED [20/26] ADD .vscode /workdir/.vscode                                                                                                                                                                                                                                                                                                     0.0s
 => CACHED [21/26] WORKDIR /workdir/workspace                                                                                                                                                                                                                                                                                                       0.0s
 => ERROR [22/26] ADD workspace /workdir/workspace                                                                                                                                                                                                                                                                                                  0.0s
------
 > [22/26] ADD workspace /workdir/workspace:
------
failed to compute cache key: "/workspace" not found: not found
```

Reply:
Issue was due to the missing workspace. In a followup #710, I made this optional (uncomment for use)